sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/InternalContext2.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.generic.core.implementation.util;
+
+import java.util.LinkedHashMap;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An {@link InternalContext} implementation that holds two key-value pairs.
+ */
+final class InternalContext2 implements InternalContext {
+    final Object key1;
+    final Object value1;
+    final Object key2;
+    final Object value2;
+
+    InternalContext2(Object key1, Object value1, Object key2, Object value2) {
+        this.key1 = key1;
+        this.value1 = value1;
+        this.key2 = key2;
+        this.value2 = value2;
+    }
+
+    @Override
+    public Object getKey() {
+        return key2;
+    }
+
+    @Override
+    public Object getValue() {
+        return value2;
+    }
+
+    @Override
+    public int count() {
+        return 2;
+    }
+
+    @Override
+    public InternalContext addData(Object key, Object value) {
+        return new InternalContext3(key1, value1, key2, value2, key, value);
+    }
+
+    @Override
+    public Optional<Object> getData(Object key) {
+        if (Objects.equals(key, key2)) {
+            return Optional.ofNullable(value2);
+        } else if (Objects.equals(key, key1)) {
+            return Optional.ofNullable(value1);
+        }
```

Comment:
If we have duplicate keys, we may return `value2` instead of `value1` with this order of evaluation and it may be different than what's expected.

Reply:
Higher value number the newer it is. This is keeping with the previous design of the linked list with a different memory footprint